tests/test_year_end_charts.py

```diff
@@ -0,0 +1,80 @@
+import abc
+import json
+import six
+import unittest
+import billboard
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Base:
+    @classmethod
+    @abc.abstractmethod
+    def setUpClass(cls):
+        pass
+
+    def testYear(self):
+        self.assertIsNotNone(self.chart.year)
+
+    def testTitle(self):
+        self.assertEqual(self.chart.title, self.expectedTitle)
+
+    def testRanks(self):
+        ranks = list(entry.rank for entry in self.chart)
+        self.assertEqual(ranks, sorted(ranks))
+
+    def testEntriesValidity(self, skipTitleCheck=False):
+        self.assertEqual(len(self.chart), self.expectedNumEntries)
+        for entry in self.chart:
+            if not skipTitleCheck:
+                self.assertGreater(len(entry.title), 0)
+            self.assertGreater(len(entry.artist), 0)
+
+    def testJson(self):
+        self.assertTrue(json.loads(self.chart.json()))
+        for entry in self.chart:
+            self.assertTrue(json.loads(entry.json()))
+
+
+class TestHot100Songs(Base, unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.chart = billboard.ChartData("hot-100-songs", year="2019")
```

Comment:
Kind of a silly request, but: is it possible to cut down on the number of tests here? The more requests the test suite makes, the more likely it is to fail because of rate-limiting or other kinds of flakiness.

Reply:
(The test suite is run daily: https://travis-ci.org/github/guoguo12/billboard-charts.)

Reply:
Sure, I understand the concern.

How about if I cut it down to three tests: `TestHot100Songs`, `TestHotCountrySongs1970`, and `TestJazzAlbumsImprints2006`?

Reply:
Addressed in [5c8c7df](https://github.com/guoguo12/billboard-charts/pull/77/commits/5c8c7dff4db13a8069d045792784a8f7ff5adea0).

Reply:
Sweet, thanks.